server/server.go

```
@@ -36,25 +41,96 @@ func (n *Node) RequestVote(ctx context.Context, req *pb.VoteRequest) (*pb.VoteRe
+	successCount := 1 // Leader tự đếm thành công vì leader đã append entry vào log
+	responseCh := make(chan bool, len(n.peers))
+
+	// Cập nhật LeaderCommit trong request theo commitIndex hiện tại của leader
+	req.LeaderCommit = int32(len(n.RaftNode.GetLog()))
+
+	// Gửi AppendEntries tới từng follower
+	for _, peer := range n.peers {
+		wg.Add(1)
+		go func(peer string) {
+			defer wg.Done()
+			conn, err := grpc.Dial("localhost:"+peer, grpc.WithInsecure())
```

golangci-lint (1.62.2)

SA1019: grpc.Dial is deprecated: use NewClient instead.  Will be supported throughout 1.x. 

(staticcheck)